internal/runners/config/config.go

```diff
@@ -18,13 +23,103 @@ type primeable interface {
 	primer.Analyticer
 }
 
-func NewConfig(prime primeable) (*Config, error) {
-	return &Config{
+func NewList(prime primeable) (*List, error) {
+	return &List{
 		out: prime.Output(),
 		cfg: prime.Config(),
 	}, nil
 }
 
-func (c *Config) Run(usageFunc func() error) error {
-	return usageFunc()
+type configData struct {
+	Key     string `locale:"key,Key"`
+	Value   string `locale:"value,Value"`
+	Default string `locale:"default,Default"`
+}
+
+type configOutput struct {
+	out     output.Outputer
+	cfg     *config.Instance
+	options []mediator.Option
+	data    []configData
+}
+
+func (c *List) Run(usageFunc func() error) error {
+	registered := mediator.AllRegistered()
+	sort.SliceStable(registered, func(i, j int) bool {
+		return registered[i].Name < registered[j].Name
+	})
+
+	var data []configData
+	for _, opt := range registered {
+		configuredValue := c.cfg.Get(opt.Name)
+		data = append(data, configData{
+			Key:     formatKey(opt.Name),
+			Value:   formatValue(opt, configuredValue),
+			Default: formatDefault(mediator.GetDefault(opt)),
+		})
+	}
+
+	c.out.Print(&configOutput{
+		out:     c.out,
+		cfg:     c.cfg,
+		options: registered,
+		data:    data,
+	})
+
+	return nil
+}
+
+func formatKey(key string) string {
+	return fmt.Sprintf("[CYAN]%s[/RESET]", key)
+}
+
+func formatValue(opt mediator.Option, value interface{}) string {
+	var v string
+	switch opt.Type {
+	case mediator.Enum:
+		return fmt.Sprintf("\"%s\"", value)
+	default:
+		v = fmt.Sprintf("%v", value)
+	}
+
+	if v == "" {
+		return "\"\""
+	}
+
+	if len(v) > 100 {
+		v = v[:100] + "..."
+	}
+
+	if value == opt.Default {
+		return fmt.Sprintf("[GREEN]%s[/RESET]", v)
+	}
+
+	return fmt.Sprintf("[BOLD][RED]%s*[/RESET]", v)
+}
+
+func formatDefault[T any](defaultValue T) string {
+	v := fmt.Sprintf("%v", defaultValue)
+	if v == "" {
+		v = "\"\""
+	}
+	return fmt.Sprintf("[DISABLED]%s[/RESET]", v)
+}
+
+func (c *configOutput) MarshalOutput(format output.Format) interface{} {
```

Comment:
Please handle structured output manually, eg. https://github.com/ActiveState/cli/blob/d43bc4921ea8fccf7f1848f4ba2e22004121e145/internal/runners/upgrade/upgrade.go#L152-L152

I'm not completely sure yet what the go-forward pattern is going to be yet, but I am sure it's going to be a lot more explicit. I'd like to move away from the magical handling.